### ISSUE TITLE
vagueness in CoC

### ISSUE THREAD

**Issue by kcpeppe** [microsoft/opensource-code-of-conduct#6]
Having attempted this myself, I appreciate the difficulties in putting a CoC together. One of the difficulties is being vague enough to cover situations that we never imaged yet not being overly vague that the CoC its self, could become an issue. I'm very certain that you had some specific idea of what you meant by the following to clause but I'm not sure that the language used is specific enough to help me understand what you have in mind.

- Disruptive behaviour
- Other conduct which could reasonably be considered inappropriate in a professional setting

For example, in technical discussions I sometimes see a disruption as being a good thing as long as it's executed thoughtfully. As for conduct which could reasonably be considered inappropriate, my opinion is that this cuts a wide path that ignores cultural diversity. Having worked in a number of European, Asian, Middle-eastern, African and American countries what I have learned is that there is a fairly broad spectrum of appropriate behaviour in professional settings.

Finally, communication always involves 2 or more people and while some things can be said with the best of intentions, there is always a risk that the message will not land well with 1 or more of the parties to the conversation. This is particularly true in multi-lingual settings where one's precision with the language being spoken maybe lacking or uneven. Just this weekend I had a conversation with a non-native English speaker who used a phrase that he thought was complimentary but to a native speaker, is offensive at best. This was not the only case where I've witness conversations take a turn for the worst when two non-native English speakers, neither spoke each others native language, both speaking in English to each other. As much as I  can hardly imagine a situation more ripe for mis-understanding, there were others witnessing the conversation that failed to understand the dynamics at work. But then, I wouldn't expect them to pick up on what was happening unless they had been living in a multi-lingual environment where you'd likely witness misunderstandings because of how different language groups learn and use English differently on a regular basis.

Finally, my apologies for pointing out issues without offering potential solutions. As I previously mentioned, I'm involved with a couple of different communities that span the globe. None of these communities have a CoC simply because we weren't able to come to agreement on wording that would address all of the concerns from these highly diverse groups. Part of the issues is that the attempts to author a CoC, in an attempt to be more concise, (less vague), drifted into controversial issues. As surprising as it might seem, agreeing to a clause that offered explicit protection for groups that are targets of discrimination (such as LGBT(...) groups) was deemed illegal in certain jurisdictions. While I argued against a CoC that drifted to a lowest common denominator, having an agreement that explicitly mentioned these groups potentially put some of our members at risk. While I don't agree with these laws, I also didn't want to place our members in any legal jeopardy either. In this case of this CoC, protection for targeted groups is weak but that might solve the problem that I've laid out.

**Comment by emmairwin** [comment]
Thanks for your thoughtful feedback @kcpeppe .

I completely understand, and have experienced the challenges, of cross-cultural collaboration, and how appropriateness of one word, or phrase may vary depending on culture and fluency of the prodominant language.  

That said, the opportunity that a code of conduct gives is to teach, and empower inclusive behaviors.   Using your example, if I use a phrase or word in French ( a language that I am poor at) , that in the language of another non-native French speaker is threatening or discrimintatory, I would want to know and improve so that I did not have this impact on someone again. It might be upsetting to learn that I had that impact on someone, but better for everyone to have that awareness.  By having no set of guidelines for participation, its more likely those who are most underrerepsented and margenlized silently leave.  There is a long history in open source of that very thing happening (diverse people leave).

In the two examples you ask about:

Disruptive behaviour
Other conduct which could reasonably be considered inappropriate in a professional setting

I have found these useful in describing the larger impact of actions that disrupt or harm healthy community collaborations.   Again the CoC is as much about generating healthy discussion and awareness as issuing consequence, and I think that lens may be helpful. I wish people talked about this positive aspect more often.
